command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -72,6 +75,17 @@ declare global {
 
     public relativeTiming(relativeTime: Temporal.Duration): Command<A>;
   }
+
+  class Sequence {
```

Comment:
I worry about exposing Sequence as a global construct that is accessible in expansions. I think there is an argument to be made that we should split at least the exposed types between expansions and sequence authoring.

Reply:
I am not sure how to limit `class Sequence` to only the `sequence authoring` side as both `expansion` and `sequence authoring` use the same preface. We could create a sequence-specific preface that is only used when expanding to a sequence. The prefaces are built when you upload a command dictionary so that means we will have to build 2 of them one for expansion and one for sequences. 

We might need a separate ticket to discuss/implement this unless there is a easier way to limit the exposure in TS? The preface makes the EDSL very flexible for us to reuse a lot of the same code and the user. Maybe this is the side effect of that though.

Reply:
I think that's fine to take this on in a different PR. Yes, we'd need to split to multiple files in order to make it work. I think the best implementation would be to keep that same implementation file and multiple global declaration files to only expose the parts we want exposed (perhaps even some shared global declarations and some differing ones).

Reply:
Would this inheritance affect the `aerie-ts-runner`? I think we can just pass more file paths into it but I wasn't sure how that works with the Global namespace of TS. 

```
#Commands
await codeRunner.executeUserCode(
      opts.edslBody,
      [],
      'Sequence',
      [],
      1000,
      [
        ts.createSourceFile('global-types.ts', opts.commandTypes, compilerTarget),
        ts.createSourceFile('command-types.ts', opts.commandTypes, compilerTarget)
        ts.createSourceFile('TemporalPolyfillTypes.ts', temporalPolyfillTypes, compilerTarget),
      ],

#Sequence
await codeRunner.executeUserCode(
      opts.edslBody,
      [],
      'Sequence',
      [],
      1000,
      [
        ts.createSourceFile('global-types.ts', opts.commandTypes, compilerTarget),
        ts.createSourceFile('sequence-types.ts', opts.commandTypes, compilerTarget)
        ts.createSourceFile('TemporalPolyfillTypes.ts', temporalPolyfillTypes, compilerTarget),
      ],
```

Reply:
Yeah, we just pass additional files with `declare global {}` blocks. I think we might need a `export {}` at the end or some such thing to ensure they're interpreted correctly.

Reply:
In fact. We could probably figure out a way to have those declaration files be static and not needed to be generated.